actionmailbox/test/controllers/ingresses/amazon/inbound_emails_controller_test.rb

```diff
@@ -0,0 +1,104 @@
+# frozen_string_literal: true
+
+require "test_helper"
+
+class ActionMailbox::Ingresses::Amazon::InboundEmailsControllerTest < ActionDispatch::IntegrationTest
+  def fixture(name)
+    file_fixture("../files/amazon/#{name}").read
+  end
+
+  def json_fixture(name)
+    JSON.parse(fixture("#{name}.json"))
+  end
+
+  setup do
+    ActionMailbox.ingress = :amazon
+    ActionMailbox.amazon = ActiveSupport::OrderedOptions.new
+    ActionMailbox.amazon.subscribed_topics = %w(
+      arn:aws:sns:eu-west-1:111111111111:example-topic
+      arn:aws:sns:eu-west-1:111111111111:recognized-topic
+    )
+    pem_url = "https://sns.eu-west-1.amazonaws.com/SimpleNotificationService-a86cb10b4e1f29c941702d737128f7b6.pem"
+    stub_request(:get, pem_url).and_return(body: fixture("certificate.pem"))
+    @inbound = json_fixture("inbound_email")
+    @inbound_s3 = json_fixture("inbound_email_s3")
+    @s3_email = fixture("s3_email.txt")
+    @invalid_signature = json_fixture("invalid_signature")
+    @valid_signature = json_fixture("valid_signature")
+    @recognized_topic = json_fixture("recognized_topic_subscription_request")
+    @unrecognized_topic = json_fixture("unrecognized_topic_subscription_request")
```

Comment:
These fixtures aren't used in most tests, most of the time. Would set them up as local variables in each test instead of ivars in setup.